File: archive/python_tools/import_additional_data.py
import pandas as pd
import sqlite3
from pathlib import Path

# Excel file path
excel_file = Path("/Users/meghakhanna/Desktop/Primary Sources/DS_Dataset.xlsx")

# Connect to SQLite database
conn = sqlite3.connect('medical_lock_hospitals.db')

try:
    # Read women admission data
    women_df = pd.read_excel(excel_file, sheet_name='Women_Admission')
    women_df.to_sql('women_admission', conn, if_exists='replace', index=False)
    print(f"Successfully imported {len(women_df)} rows into women_admission table")

    # Read troops data
    troops_df = pd.read_excel(excel_file, sheet_name='Troops')
    troops_df.to_sql('troops', conn, if_exists='replace', index=False)
    print(f"Successfully imported {len(troops_df)} rows into troops table")

except Exception as e:
    print(f"Error occurred: {str(e)}")

finally:
    conn.close()

File: archive/scripts_migrations/import_additional_data.R
library(readxl)
library(DBI)
library(RSQLite)

# Excel file path
excel_file <- "/Users/meghakhanna/Desktop/Primary Sources/DS_Dataset.xlsx"

# Connect to SQLite database
conn <- dbConnect(RSQLite::SQLite(), "medical_lock_hospitals.db")

tryCatch({
  # Read women admission data
  women_df <- read_excel(excel_file, sheet = "women_admission")
  dbWriteTable(conn, "women_admission", women_df, overwrite = TRUE)
  cat(sprintf("Successfully imported %d rows into women_admission table\n", nrow(women_df)))

  # Read troops data
  troops_df <- read_excel(excel_file, sheet = "troops_admission")
  dbWriteTable(conn, "troops", troops_df, overwrite = TRUE)
  cat(sprintf("Successfully imported %d rows into troops table\n", nrow(troops_df)))

}, error = function(e) {
  cat("Error occurred:", conditionMessage(e), "\n")
}, finally = {
  dbDisconnect(conn)
})